elasticsearch/src/main/java/org/hibernate/search/elasticsearch/impl/ElasticsearchIndexManager.java

```diff
@@ -293,15 +297,19 @@ public void performOperations(List<LuceneWork> workList, IndexingMonitor monitor
 
 	@Override
 	public void performStreamOperation(LuceneWork singleOperation, IndexingMonitor monitor, boolean forceAsync) {
-		if ( singleOperation == FlushLuceneWork.INSTANCE ) {
+		if ( singleOperation instanceof FlushLuceneWork ) {
 			requestProcessor.awaitAsyncProcessingCompletion();
 		}
-		else {
-			BackendRequest<?> request = singleOperation.acceptIndexWorkVisitor( visitor, monitor );
 
-			if ( request != null ) {
+		BackendRequest<?> request = singleOperation.acceptIndexWorkVisitor( visitor, monitor );
+
+		if ( request != null ) {
+			if ( forceAsync ) {
 				requestProcessor.executeAsync( request );
 			}
+			else {
+				requestProcessor.executeSync( Collections.<BackendRequest<?>>singletonList( request ) );
```

Comment:
So, here, we're switching the default behavior (when `forceAsync` is `false`) from asynchronous to synchronous.

From what I can see in `org.hibernate.search.indexes.spi.IndexManager.performStreamOperation(LuceneWork, IndexingMonitor, boolean)`, the behavior when `forceAsync` is `false` "will depend on the backend configuration". So I guess we could make it async by default if we wanted to.

I must admit I don't quite grasp the implications of that change, though. The whole concept of being able to execute "stream" operations asynchronously or synchronously seems rather blurry to me, so I must be missing something.

@Sanne , what do you think?